DatadogSessionReplay/Sources/Utilities/UIImage+Scaling.swift

```diff
@@ -0,0 +1,74 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import UIKit
+
+extension UIImage {
+    /// Scales down the image to an approximate file size in bytes.
+    ///
+    /// - Parameter desiredSizeInBytes: The target file size in bytes.
+    /// - Returns: A Data object representing the scaled down image as PNG data.
+    ///
+    /// This function takes the desired file size in bytes as input and scales down the image iteratively
+    /// until the resulting PNG data size is less than or equal to the specified target size.
+    ///
+    /// Note: The function will return the original image data if it is already smaller than the desired size,
+    /// or if it fails to generate a smaller image.
+    ///
+    /// Example usage:
+    ///
+    /// let originalImage = UIImage(named: "exampleImage")
+    /// let desiredSizeInBytes = 10240 // 10 KB
+    /// if let imageData = originalImage?.scaledDownToApproximateSize(desiredSizeInBytes) {
+    ///     // Use the scaled down image data.
+    /// }
+    func scaledDownToApproximateSize(_ desiredSizeInBytes: Int) -> Data {
+        guard let imageData = pngData() else {
+            return Data()
+        }
+        guard imageData.count > desiredSizeInBytes else {
+            return imageData
+        }
+        var scaledImage = scaledImage(by: sqrt(CGFloat(desiredSizeInBytes) / CGFloat(imageData.count)))
```

Comment:
It normalises to the scale to average side of rectangle instead of scale of the whole area. If you have: `a2 * b2 = S1` and `a2 * b2 = S2`. S1/S2 is a percentage of areas, but if you use `sqrt()` it gives you scale of average side (calculating the side of the square). From experiments calculating the scaling like this was giving better initial scaled image (closer to actual desired size in bytes).

Reply:
Got it 👍, it's cleaner now with this comment, but still magical in the code 😉. We could enhance it by adding `//` comment to explain that `sqrt()` is to _"approximate the side of square with given area"_. Also if it's based on experiments, it might make sense to add this information too - so that if changing this code we know the full context.